pykafka/balancedconsumer.py

```diff
@@ -509,6 +525,7 @@ def _consumers_changed(self, consumers):
         if self._setting_watches:
             return
         log.debug("Rebalance triggered by consumer change")
+        self._add_self()
```

Comment:
If the zookeeper connection dies, the consumer's ID is removed from ZK's registry. When the zookeeper connection is regained, the consumer needs to actively add itself to that registry. Since we can't count on the zookeeper connection failing in a particular piece of code, this seems like the best place to handle it. If we have an active ZK connection, this function will be called by its watch and `_add_self` will happen.